### ISSUE TITLE
Hangs still present when using msys2 shell

### ISSUE THREAD

**Issue by jrick** [Masterminds/glide#203]
I'm not certain that this is still an issue with glide itself, but I think it's worthwhile documenting it at least somewhere.

The latest fixes on master do make glide work correctly for me when I run it from a cmd.exe or powershell window, but if I start the msys2 shell (which uses the mintty terminal emulator) it ends up hanging after the first "scanning X for dependencies" log.  CPU usage also ramps up so it's probably spinning endlessly in a loop.

Another interesting case is that if I start a cmd.exe or powershell session (even with the powershell ISE) from the msys2 shell, glide still hangs.  Even weirder is that if I start msys2's bash from a cmd.exe or powershell window, everything works correctly.  I don't think this issue is inherent to the mintty terminal emulator itself (otherwise why would it affect the powershell ISE?) so that msys2 session is probably setting some environment variables that end up breaking glide (since these are inherited by the child processes).

I typically use powershell so this isn't a dealbreaker for me but I know others will only want to use something like msys2.

**Comment by thinkong** [comment]
seems to be a similar issue that I added https://github.com/Masterminds/glide/issues/202

**Comment by mattfarina** [comment]
@jrick can you fill me in on your environment (e.g., `go env`)?

I posted a fix at 285779832ef12946ddc451a81b0f2fefea42b706 that could cause an infinite loop on scanning in some situations (most notably Windows). Is it any better using the tip of master now?

**Comment by jrick** [comment]
Seems to be fixed for me on master now (glide up works no matter what drive my GOPATH is on, and is working in all shells).  Thanks!

FWIW: my `go env` environment variables were the same in both shells:

```
set GOARCH=amd64
set GOBIN=
set GOEXE=.exe
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=D:\Go
set GORACE=
set GOROOT=C:\Go
set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
set GO15VENDOREXPERIMENT=
set CC=gcc
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0
set CXX=g++
set CGO_ENABLED=1
```
